nsdb-http/src/main/scala/io/radicalbit/nsdb/web/routes/QueryApi.scala

```diff
@@ -99,42 +100,92 @@ trait QueryApi {
       new ApiResponse(code = 400, message = "statement is invalid")
     ))
   def queryApi()(implicit logger: LoggingAdapter, format: Formats): Route = {
+
+    def executeSelectStatement(queryBody: QueryBody, statement: SelectSQLStatement) = {
+      onComplete(readCoordinator ? ExecuteStatement(statement)) {
+        case Success(SelectStatementExecuted(_, values)) =>
+          complete(
+            HttpEntity(ContentTypes.`application/json`,
+                       write(SelectQueryResponse(values, queryBody.parsed.map(_ => statement)))))
+        case Success(SelectStatementFailed(_, reason, MetricNotFound(_))) =>
+          complete(HttpResponse(NotFound, entity = reason))
+        case Success(SelectStatementFailed(_, reason, _)) =>
+          complete(HttpResponse(InternalServerError, entity = reason))
+        case Success(r) =>
+          logger.error("unknown response received {}", r)
+          complete(HttpResponse(InternalServerError, entity = "unknown response"))
+        case Failure(ex) =>
+          logger.error(ex, s"error while trying to execute $statement")
+          complete(HttpResponse(InternalServerError, entity = ex.getMessage))
+      }
+    }
+
+    def executeDeleteStatement(statement: DeleteSQLStatement) = {
+      onComplete(writeCoordinator ? ExecuteDeleteStatement(statement)) {
+        case Success(_: DeleteStatementExecuted) => complete("Ok")
```

Comment:
Can the user benefit to receive a message about the number of records cleared? Action duration either. What do you think?

Reply:
1. action duration => I don't know... I've always found it redundant. If we consider the protocol overhead insignificant, the HTTP call duration is the action duration itself.

2. Number of records cleared => Lucene gives an API to get the total number of deleted documents for an index but not the total number of deleted document for a given operation. It would be possible to have it by executing a count query with the provided where condition before executing the actual delete operation.
Basically, if we really need it, we can add it, but with a cost.